projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts

```diff
@@ -350,335 +243,51 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
-    id: string;
-    name: string;
-    authorName: string;
-    authorInstitution: string;
-    description: string;
-    license: string;
-    datasetPath: string;
-    ebvClass: string;
-    ebvName: string;
-}
-
-interface EbvHierarchy {
-    providerId: UUID;
-    tree: EbvTree;
-}
-
-interface EbvTree {
-    fileName: string;
-    title: string;
-    spatialReference: string;
-    groups: Array<EbvTreeSubgroup>;
-    entities: Array<EbvTreeEntity>;
-    timeCoverage: TimeCoverageRegular | TimeCoverageList;
-    colorizer: ColorizerDict;
-}
-
-interface TimeCoverageRegular {
-    type: 'regular';
-    start: number;
-    end: number;
-    step: TimeStepDict;
-}
-
-interface TimeCoverageList {
-    type: 'regular';
-    timeStamps: [number];
-}
-
-interface EbvTreeSubgroup {
-    name: string;
-    title: string;
-    description: string;
-    dataType?: 'U8' | 'U16' | 'U32' | 'U64' | 'I8' | 'I16' | 'I32' | 'I64' | 'F32' | 'F64';
-    dataRange?: [number, number];
-    groups: Array<EbvTreeSubgroup>;
-}
-
-interface EbvTreeEntity {
-    id: number;
-    name: string;
-    description: string;
-}
-
-interface EbvDatasetId {
-    fileName: string;
-    groupNames: Array<string>;
-    entity: number;
-}
-
-function guessDataRange(colorizer: Colorizer): {min: number; max: number} {
-    let min = Number.MAX_VALUE;
-    let max = -Number.MAX_VALUE;
-
-    for (const breakpoint of colorizer.getBreakpoints()) {
-        min = Math.min(min, breakpoint.value);
-        max = Math.max(max, breakpoint.value);
-    }
-
-    return {min, max};
-}
-
-function timeStartFromCoverage(timeCoverage: TimeCoverageRegular | TimeCoverageList): number {
-    if (timeCoverage.type === 'regular') {
-        const timeCoverageRegular = timeCoverage as TimeCoverageRegular;
-        return timeCoverageRegular.start;
-    } else if (timeCoverage.type === 'list') {
-        const timeCoverageList = timeCoverage as TimeCoverageList;
-        return timeCoverageList.timeStamps[0];
-    } else {
-        throw Error('unsupported time coverage type ' + timeCoverage.type);
-    }
-}
-
-function timeEndFromCoverage(timeCoverage: TimeCoverageRegular | TimeCoverageList): number {
-    if (timeCoverage.type === 'regular') {
-        const timeCoverageRegular = timeCoverage as TimeCoverageRegular;
-        return timeCoverageRegular.end;
-    } else if (timeCoverage.type === 'list') {
-        const timeCoverageList = timeCoverage as TimeCoverageList;
-        return timeCoverageList.timeStamps[timeCoverageList.timeStamps.length - 1];
-    } else {
-        throw Error('unsupported time coverage type ' + timeCoverage.type);
-    }
+interface EbvDatasetResponse {
+    code: number;
+    data: [
+        {
+            title: string;
+            ebv: {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                ebv_class: string;
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                ebv_name: string;
```

Comment:
I guess we have to rename this in our backend to adhere to our camelCase default.
Or is this from iDivs web service?

Reply:
this is directly from the geobon API as you suggested